libshortfin/python/shortfin_apps/llm/components/generate.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2024 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import asyncio
+import io
+import logging
+
+import shortfin as sf
+import shortfin.array as sfnp
+
+# TODO: Have a generic "Responder" interface vs just the concrete impl.
+from shortfin.interop.fastapi import FastAPIResponder
+
+from .io_struct import GenerateReqInput
+from .messages import InferenceExecRequest, InferencePhase
+from .service import GenerateService
+from .tokenizer import Encoding
+
+logger = logging.getLogger(__name__)
+
+
+class GenerateItemProcess(sf.Process):
+    """Process instantiated for each generation sequence.
+
+    This process breaks the sequence into individual inference and sampling
+    steps, submitting them to the batcher and marshaling incremental/final
+    results.
+    """
+
+    def __init__(
+        self,
+        client: "ClientGenerateBatchProcess",
+        gen_req: GenerateReqInput,
+        index: int,
+        input_token_ids: list[int],
+    ):
+        super().__init__(fiber=client.fiber)
+        self.client = client
+        self.gen_req = gen_req
+        self.index = index
+        self.input_token_ids = input_token_ids
+        self.result_token_ids: list[int] = []
+
+    async def run(self):
+        exec = InferenceExecRequest(InferencePhase.PREFILL, self.input_token_ids)
+        try:
+            self.client.batcher.submit(exec)
+            await exec.done
+
+            # Prefill result.
+            token = sfnp.argmax(exec.result_logits)
+            token_int = token.items[0]
+
+            # Decode loop.
+            # TODO: Use correct eot token from config.
+            exec.start_position = len(self.input_token_ids) - 1
+            while token_int != 128001:
```

Comment:
Just tagging to remind in case you missed the TODO

Reply:
Will come back through. There are a lot of things that need to come from config and are not right now.